src/components/KCatalog/KCatalog.vue

```diff
@@ -360,20 +373,37 @@ const emit = defineEmits<{
 const slots = useSlots()
 
 const catalogId = useUniqueId()
-const defaultFetcherProps = {
-  page: 1,
-  pageSize: 15,
-  query: '',
+
+const getInitialPageSize = (): number => {
+  const initialPageSize = props.paginationPageSizes?.[0]
+
+  if (initialPageSize) {
+    return initialPageSize
+  }
+
+  return DEFAULT_PAGE_SIZE
 }
 
 const data = ref<CatalogItem[]>([])
 const total = ref<number>(0)
 const filterQuery = ref<string>('')
 const page = ref<number>(1)
-const pageSize = ref<number>(15)
+const pageSize = ref<number>(getInitialPageSize())
+const offset = ref<string | null>(null)
+const offsets = ref<Array<any>>([])
+const previousOffset = computed((): string | null => offsets.value[page.value - 1])
```

Comment:
```suggestion
const previousOffset = computed((): string | null => offsets.value[page.value - 1])
```

will it be a problem if `offsets.value` is null?